### ISSUE TITLE
realm.executeTransactionAsync() is NOT always executed in Realm 5.0/5.0.1

### ISSUE THREAD

**Issue by aksinghnet** [realm/realm-java#5900]
#### Goal

I am trying to load an object in a background thread using realm.executeTransactionAsync()

#### Expected Results

realm.executeTransactionAsync() should always execute, thus loading the object from realm to UI

#### Actual Results

realm.executeTransactionAsync() is executed only sometimes, and other times, it is not executed

#### Code Sample

```kotlin
println("activity.realm closed = ${activity.realm.isClosed}") // always printed "false"

activity.realm.executeTransactionAsync { realmAsync ->
      println("loadOffline inAsync tickers INSIDE") // printed sometimes

      val tickers = realm.where(CoinTickerList::class.java).equalTo("uuid", "MY_ID").findFirst()!!

      println("loadOffline inAsync tickers.size = ${tickers.size}") // printed sometimes
}
```

Sometimes, I see the "loadOffline inAsync tickers INSIDE" in the output and sometimes I do not see this output. This function is called when app is opened to load the data from realm to UI. I am checking by opening the app multiple times, but in sometimes, I do not see this output in logcat.

#### Version of Realm and tooling
Realm version(s): 5.0.1 at present [It was working fine before when my earlier objects were compiled using 4.3.3 with Android Gradle plugin 3.0.1. Then I updated Realm to 5.0.0, later updated Android Gradle plugin to 3.1, then created a new class of RealmObject and getting problem after that.]

Realm sync feature enabled: No

Android Studio version: 3.1.1 (latest)

Which Android version and device: Android 8.1 (Oreo) on One Plus 3T

**Comment by Zhuinden** [comment]
Maybe `!!` fails because the object is not in the Realm,

**Comment by aksinghnet** [comment]
Hi @Zhuinden, thanks for the quick response.

I strongly believe it is not the case because this object is created using initialdata().
Also, I have seen that `!!` results in an exception if it is null, which I am not getting here.

```kotlin
val realmConfig = RealmConfiguration.Builder()
.initialData { realm ->
        CoinTickerList.createCopyToRealm(realm, "MY_ID")
}
```

Moreover, please note that the first line in the async block is a print statement, and this statement is not printed every time. Could you suggest some other idea or solution that I can try?

**Comment by aksinghnet** [comment]
Hi all, thanks for your responses. I finally found the issue and wish to share the solution here. :)

I found I was running realm.executeTransactionAsync() in a service, which was executing it multiple times, thus multiple realm async tasks and multiple connections. And the task was taking a long time to complete, so it seemed that the async task was not running, but it was running after a while.

So the conclusion/lesson here is to avoid realm.executeTransactionAsync() for long tasks and keep its instance to close/cancel it upon exit (onStop or onDestroy) of the activity or the service.

Thank you.
